pandas/core/frame.py

```diff
@@ -1133,9 +1130,9 @@ def to_string(
     def to_string(
         self,
         buf: FilePath | WriteBuffer[str],
-        columns: Sequence[str] | None = ...,
+        columns: Axes | None = ...,
         col_space: int | list[int] | dict[Hashable, int] | None = ...,
-        header: bool | Sequence[str] = ...,
+        header: bool | list = ...,
```

Comment:
avoid invariant collection types (list, dict) in argument positions, in favor of covariant types like Mapping or Sequence;

Reply:
Same issue `Sequence` allows a regular string to be passed, which is invalid.

Reply:
Changing it to `list[str]`